src/omnidirectional_controller.cpp

```diff
@@ -229,34 +231,47 @@ CallbackReturn OmnidirectionalController::on_activate(
   // register handles
   registered_wheel_handles_.reserve(wheel_names_.size());
   for (const auto & wheel_name : wheel_names_) {
-    std::string interface_name = wheel_name + "/" + HW_IF_VELOCITY;
 
-    const auto state_handle = std::find_if(state_interfaces_.cbegin(), state_interfaces_.cend(),
-      [&interface_name](const auto & interface) {
-        return interface.get_name() == interface_name;
-      });
+    const hardware_interface::LoanedStateInterface* state_handle = nullptr;
```

Comment:
Could you, please, explain the need of this raw pointer here? I didn't get the idea

Reply:
The reason for using a raw pointer (state_handle) in this context is to manually manage the lifetime of the pointer. Since the pointer is only used as part of this function and its validity depends on the !odom_params_.open_loop condition, I wanted to avoid dynamically allocating memory for the LoanedStateInterface object and instead use a pointer directly to the object. 

Using std::shared_ptr would be another solution but would require modifying the WheelHandle structure if I am to believe the compilation errors I got during my tests. 


However, as my expertise with CPP is not as extensive as I would like, I am open to other methods.

Reply:
I got it, but I thought this approach cumbersome. Here is my suggestion

Transform this ([lines 61 to 67](https://github.com/mateusmenezes95/omnidirectional_controllers/blob/humble/include/omnidirectional_controllers/omnidirectional_controller.hpp#L61-L67))

```cpp
  struct WheelHandle {
    std::reference_wrapper<const hardware_interface::LoanedStateInterface> velocity_state;
    std::reference_wrapper<hardware_interface::LoanedCommandInterface> velocity_command;
  };

  std::vector<std::string> wheel_names_;
  std::vector<WheelHandle> registered_wheel_handles_;
```
in
```cpp
  using StateInterfaceReferenceWrapper = std::reference_wrapper<const hardware_interface::LoanedStateInterface>;
  using CommandInterfaceReferenceWrapper = std::reference_wrapper<hardware_interface::LoanedCommandInterface>;

  std::vector<std::string> wheel_names_;
  std::vector<StateInterfaceReferenceWrapper> registered_wheel_state_ifs_;
  std::vector<CommandInterfaceReferenceWrapper> registered_wheel_cmd_ifs_;
```

With this modification we can thus work with the state and command interfaces handles independently. What do you think?

Reply:
It's clearly the simplest and clearest - you're absolutely right. As I was trying not to alter your initial work, I wanted to avoid making any changes to the hpp. If you don't see any problem with that, then I'll get down to it tomorrow.

Reply:
Nice! There is no problem for me in changing the hpp. You can modify your wishes

Reply:
Updated and seems to work fine to me. Does it work for you?